File: PJC/Cwiczenia1/Zadanie1.cpp
#include <iostream>
#include <vector>

using namespace std;

auto countChars(vector<char> letters, char a)
{
    int count = 0;
    for(auto c : letters)
        if(a == c)
            count++;
    return count;
}

auto countChars(string letters, char a) -> int
{
    int count = 0;
    for(auto c : letters)
        if(c == a)
            count++;
    return count;
}

auto main() -> int
{
    cout << countChars(std::vector<char>{'a', 'l', 'a'}, 'a') << '\n'
              << countChars("", 'a') << '\n'
              << countChars(std::string{'k', 'a', 'j', 'a', 'k'}, 'a') << '\n'
              << countChars(std::string("kajak"), 'k') << '\n'
              << countChars(std::vector<char>{'k', 'a', 'j', 'a', 'k'}, 'j') << '\n'
              << countChars("kajak", 'x') << '\n';

    return 0;
}

File: PPJ/src/FirstSemester/Cwiczenia27/Zadanie1.java
package PPJ.FirstSemester.Cwiczenia27;
import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;

public class Zadanie1
{
    static boolean letters(char c)
    {
        return (c >= 65 && c <= 90 || c >= 97 && c <= 122 );
    }
    static boolean specialChars(char c)
    {
        return (c == '.' || c == ' ' || c == '/' || c == '\n' || c == '(');
    }
    public static void main(String[] args) throws FileNotFoundException
    {
        File file = new File("/Users/michallejza/Downloads/PJWSTK/src/Cwiczenia/Cwiczenia27/Zadanie1.java");
        Scanner scanFile = new Scanner(file);
        int count = 0;
        while (scanFile.hasNextLine())
        {
            String line = scanFile.nextLine();
            for(int i = 0; i < line.length() - 1; i++)
                if(letters(line.charAt(i)) && specialChars(line.charAt(i + 1)))
                    count++;
        }
        System.out.println(count);
    }
}
